provider/core/src/hello_world.rs

```diff
@@ -21,7 +19,11 @@ use writeable::Writeable;
 /// A struct containing "Hello World" in the requested language.
 #[derive(Debug, PartialEq, Clone, Yokeable, ZeroFrom)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
+#[cfg_attr(
+    any(feature = "deserialize_json", feature = "datagen"),
```

Comment:
Suggestion/Issue: Can you gate this on just `serde_json` since it doesn't have to do with deserialization, and it's hypothetically possible that someone would want this impl without deserialize_json (such as for testing purposes)?

Reply:
`serde_json` is not a feature

Reply:
Isn't it an implicit feature?

The issue is that this impl really need not be tied to `"deserialize_json"`. I skirted around this issue before with the custom JSON impl. Maybe that's just me wanting purity where it isn't necessary; but, we're just trading one type of purity for another. Do we care more about purity of features or purity of serde impls?